packages/remote-feature-flag-controller/src/remote-feature-flag-controller.test.ts

```diff
@@ -0,0 +1,347 @@
+      const clientConfigApiService = buildClientConfigApiService();
+      const controller = createController({
+        clientConfigApiService,
+        disabled: true,
+      });
+
+      controller.enable();
+      await controller.updateRemoteFeatureFlags();
+      expect(clientConfigApiService.fetchRemoteFeatureFlags).toHaveBeenCalled();
+    });
+
+    it('preserves cached flags and returns cached data when disabled', async () => {
+      const clientConfigApiService = buildClientConfigApiService();
+      const controller = createController({
+        clientConfigApiService,
+        disabled: false,
+        state: { remoteFeatureFlags: MOCK_FLAGS, cacheTimestamp: 0 },
+      });
+
+      // Verify cache is preserved
+      expect(controller.state.remoteFeatureFlags).toStrictEqual(MOCK_FLAGS);
+
+      // Disable the controller
+      controller.disable();
+
+      // Verify disabled behavior,
+      // cache should be preserved, but updateRemoteFeatureFlags should return empty array
+      await controller.updateRemoteFeatureFlags();
+      expect(controller.state.remoteFeatureFlags).toStrictEqual(MOCK_FLAGS);
+      expect(controller.state.remoteFeatureFlags).toStrictEqual(MOCK_FLAGS);
+    });
+  });
+});
+
+type RootAction = RemoteFeatureFlagControllerActions;
+type RootEvent = RemoteFeatureFlagControllerStateChangeEvent;
+
+/**
+ * Creates and returns a root controller messenger for testing
+ * @returns A controller messenger instance
+ */
+function getRootControllerMessenger(): ControllerMessenger<
+  RootAction,
+  RootEvent
+> {
+  return new ControllerMessenger<RootAction, RootEvent>();
+}
+
+/**
+ * Creates a restricted controller messenger for testing
+ * @param rootMessenger - The root messenger to restrict
+ * @returns A restricted controller messenger instance
+ */
+function getControllerMessenger(
+  rootMessenger = getRootControllerMessenger(),
+): RemoteFeatureFlagControllerMessenger {
+  return rootMessenger.getRestricted({
+    name: controllerName,
+    allowedActions: [],
+    allowedEvents: [],
+  });
+}
+
+/**
+ * Builds a mock client config API service for testing
+ * @param options - The options object
+ * @param options.remoteFeatureFlags - Optional feature flags data to return
+ * @param options.cacheTimestamp - Optional timestamp to use for the cache
+ * @param options.error - Optional error to simulate API failure
+ * @returns A mock client config API service
+ */
+function buildClientConfigApiService({
+  remoteFeatureFlags,
+  cacheTimestamp,
+  error,
+}: {
+  remoteFeatureFlags?: FeatureFlags;
+  cacheTimestamp?: number;
+  error?: Error;
+} = {}): AbstractClientConfigApiService {
+  return {
+    fetchRemoteFeatureFlags: jest.fn().mockImplementation(() => {
+      if (error) {
+        return Promise.reject(error);
+      }
+      return Promise.resolve({
+        error: false,
+        message: 'Success',
+        statusCode: '200',
+        statusText: 'OK',
```

Comment:
Nit: These properties were removed I believe? They should not be needed for this mock
```suggestion
```

Reply:
tracked here 040a4950fae4f654602641cb88d9fc536de317b8 👍🏻

Reply:
Looks like you found two other examples, but this one remains here

Reply:
Oh oh good catch ! 37d0bcc0390fe40e04a285856d0f7e3e03022b0d